builtin/checkout--helper.c

```diff
@@ -0,0 +1,1288 @@
+#include "builtin.h"
```

Comment:
The commit introducing `git checkout--helper` is jarring in its size compared to the previous ones. I think the commit message deserves some love to assist the reviewer with how this helper is organized.

Reply:
+100